xwiki-platform-core/xwiki-platform-livetable/xwiki-platform-livetable-ui/src/main/resources/XWiki/LiveTableResultsMacros.xml

```diff
@@ -739,8 +739,11 @@
  * NOTE: This macro uses variables defined in livetable_filterProperty . It was not meant to be used alone.
  *#
 #macro (livetable_filterStringListProperty)
-  #set ($whereSql = "${whereSql} and concat('|', concat(upper(${safe_tableAlias}.textValue), '|')) like upper(?)")
-  #set ($discard = $whereParams.add("%|$filterValue|%"))
+  #set ($filterValues = $request.getParameterValues($colname))
+  #foreach ($filterValue in $filterValues)
+    #set ($whereSql = "${whereSql} and concat('|', concat(upper(${safe_tableAlias}.textValue), '|')) like upper(?)")
```

Comment:
In other words, you don't have to add the UI that lets users switch, but add support right here for a configurable join mode, with the default set to "AND". Feel free to copy code from https://github.com/phenotips/phenotips/blob/phenotips-1.3/components/navigation/ui/src/main/resources/PhenoTips/LiveTableResults.xml#L562-L566